### PR TITLE
(724) Move crn search to start of apply journey

### PR DESCRIPTION
**Pull request by richpjames**
# Context
Because the CRN is a unique identifier - we should have the person search sitting seperately from the application process. 
This means that we have to remove the CRN lookup and confirmation from the regular 'Apply' form flow and have it follow the pattern of the BookingController 'new' method.

[Trello ticket](https://trello.com/c/fTV8f4cL/724-move-crn-search-to-the-start-of-the-journey)

### CODE REVIEW

File: server/controllers/apply/applicationsController.test.ts [hunk 1]
```diff
@@ -39,12 +51,95 @@ describe('applicationsController', () => {
...
         pageHeading: 'Apply for an Approved Premises (AP) placement',
       })
     })
   })
 
+  describe('new', () => {
+    describe('If there is a CRN in the flash', () => {
+      it('it should render the start of the application form', async () => {
+        const person = personFactory.build()
+        personService.findByCrn.mockResolvedValue(person)
```
Comment (pezholio) [inline server/controllers/apply/applicationsController.test.ts:64]: Might make sense to have this in a `beforeEach` step, as we have in the booking test now

File: server/controllers/apply/applicationsController.ts [hunk 2]
```diff
@@ -13,17 +16,47 @@ export default class ApplicationsController {
...
+        errors,
+        errorSummary,
+        ...userInput,
+      })
     }
   }
 
   create(): RequestHandler {
     return async (req: Request, res: Response) => {
       const uuid = await this.applicationService.createApplication(req.user.token)
```
Comment (pezholio) [inline server/controllers/apply/applicationsController.ts:57]: I think we'll want to pass the user to the application here too, but that can maybe come later

**Review by pezholio — APPROVED** [review]
LGTM 👍

File: server/controllers/apply/applicationsController.test.ts (hunk 1 above)
Comment (richpjames) [reply server/controllers/apply/applicationsController.test.ts:64]: Absolutely!

**Review by pezholio — APPROVED** [review]
LGTM 👍

**Pull request merged**